### DISCUSSION TITLE
Creating Radius Profiles

### DISCUSSION THREAD

**Discussion by munfai-79** [Art-of-WiFi/UniFi-API-client#238] [General]
I have a requirement to create a new Radius profile for several hundred sites... 

Does anyone know of a way to do this using the API ? The profile is identical across all sites - same name, same auth server IP/port, same key

I see there is a function called list_radius_profile in the API, however there is no function to create/set one. 

Thank you in advance

**Comment by sonnicmedia** [comment]
Hi, I have a similar request but to create Port Profiles across multiple hundreds of sites. Any possibility to do this with API?
I got to "list the profiles" but seems like there is no way to Crate/Modify.
Thanks

**Reply by malle-pietje** [reply]
The route for creating ethernet port profiles has not (yet) been implemented. You can use the custom method and use the following info:
- URL is https://controller:8443/api/s/site-id/rest/portconf
- method is POST
- payload looks like so in JSON format (you'll need to convert that to a PHP associative array when passing that to the API using this API Client):
```json
{
  "name": "Testing123",
  "op_mode": "switch",
  "poe_mode": "auto",
  "port_security_enabled": false,
  "port_security_mac_address": [],
  "native_networkconf_id": "zzzzzzzzzzzzzzzzzzzzz",
  "tagged_vlan_mgmt": "auto",
  "multicast_router_networkconf_ids": [],
  "setting_preference": "auto",
  "dot1x_idle_timeout": 300,
  "voice_networkconf_id": "",
  "stp_port_mode": true,
  "lldpmed_enabled": true,
  "fec_mode": "disabled",
  "qos_profile": {
    "qos_policies": [],
    "qos_profile_mode": "custom"
  },
  "autoneg": true,
  "dot1x_ctrl": "force_authorized",
  "egress_rate_limit_kbps_enabled": false,
  "isolation": false,
  "lldpmed_notify_enabled": false,
  "port_keepalive_enabled": false,
  "stormctrl_bcast_enabled": false,
  "stormctrl_mcast_enabled": false,
  "stormctrl_ucast_enabled": false,
  "excluded_networkconf_ids": []
}
```
Not sure whether the `native_networkconf_id` value is required, you would have to test that. Please share your experience here.

**Reply by malle-pietje** [reply]
Feel free to submit a PR to implement a method/function for this. Should be pretty similar to the `create_dynamicdns()` method.

**Reply by sonnicmedia** [reply]
Thank you @malle-pietje this is the exact info I needed. Seems like native_networkconf_id gets randomly assigned.

What would be function to create network on a site?

**Reply by malle-pietje** [reply]
@sonnicmedia that is supported by an existing function. Please search the source to find it.

**Comment by malle-pietje** [comment]
@munfai-79 For your use-case the URL is structured like so:
https://controller:8443/api/s/site-id/rest/radiusprofile

HTTP request is POST.

Payload looks roughly like this:
```json
{
  "attr_no_delete": false,
  "attr_hidden_id": "",
  "name": "Testing123",
  "use_usg_auth_server": false,
  "auth_servers": [
    {
      "ip": "192.168.0.100",
      "port": 1812,
      "x_secret": "blablabla"
    }
  ],
  "acct_servers": [],
  "vlan_enabled": true,
  "vlan_wlan_mode": "disabled",
  "accounting_enabled": false,
  "interim_update_enabled": true,
  "interim_update_interval": "3600",
  "x_ca_crts": []
}
```
With this info you can use the custom method or create a PR and contribute a method/function similar to the above mentioned one.
